lib/quoil/cli.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Quoil.CLI do
```

Comment:
Understandable about getting code working. Useful code is better than perfect code that is never run :smile: 

I think that it is clear to move the termination logic into `parse_args` and comment it saying that it may terminate the program. The main thing is to make code obvious and unsurprising. To me anyway, having `parse_args` complain and blow up on bad args is fine and makes a lot of sense. Those kind of calls can get quite subjective based on personal preference though, and there isn't any right answer.

Alternatively, you could go for a ["single function exit"](https://stackoverflow.com/questions/4838828/why-should-a-function-have-only-one-exit-point) style (another subjective opinion...) and have something that looks like this:

```elixir
case Quoil.ArgsProcessor.parse_args do
  {:ok, ip_to_ping, switches, log_file_name} ->
    run_ping(ip_to_ping, switches, log_file_name)
    |>  ...
  {:info, message} ->
    IO.puts(:stdio, message)
    System.halt(0)
  {:error, message} ->
    IO.puts(:stderr, message)
    System.halt(1)
end
```

So with that approach, `parse_args` returns a tuple that can be matched on. The nice thing about this is that `parse_args` is responsible for handling all the messages, and logic, and the main app just carries on, or exits depending on it.

Note that the error case is outputting to stderr rather than stdin. Also, it's exiting with status 1 as opposed to using `:abort`. `:abort` will cause a core dump and is meant for unexpected crashes and other fatal errors, rather than bad user input.